lib/matplotlib/tests/test_colorbar.py

```diff
@@ -191,6 +191,66 @@ def test_gridspec_make_colorbar():
     plt.subplots_adjust(top=0.95, right=0.95, bottom=0.2, hspace=0.25)
 
 
+@image_comparison(baseline_images=['colorbar_join',
+                                   'colorbar_join_frac', ],
+                  extensions=['png'], remove_text=True,
+                  savefig_kwarg={'dpi': 40})
+def test_join_colorbar():
+    data = np.arange(1200).reshape(30, 40)
+
+    # Jet is a LinearSegmentedColormap
+    cmap1 = plt.get_cmap('viridis', 5)
+    cmap2 = plt.get_cmap('jet', 5)
+
+    # This should be a listed colormap.
+    cmap = cmap1.join(cmap2)
+
+    plt.figure()
+    plt.pcolormesh(data, cmap=cmap)
+    plt.colorbar(orientation='vertical')
+
+    # Use the 'frac_self' kwarg for the listed cmap
+    cmap = cmap1.join(cmap2, frac_self=0.7, N=50)
+
+    plt.figure()
+    plt.pcolormesh(data, cmap=cmap)
+    plt.colorbar(orientation='vertical')
+
+
+@image_comparison(baseline_images=['colorbar_truncate',
+                                   'colorbar_trunc-getitem',
+                                   'colorbar_trunc-getitem-int',
+                                   'colorbar_trunc-getitem-int-1jN', ],
+                  extensions=['png'], remove_text=True,
+                  savefig_kwarg={'dpi': 40})
+def test_truncate_colorbar():
+    data = np.arange(1200).reshape(30, 40)
+
+    cmap = plt.get_cmap('viridis', 32).truncate(0.2, 0.7)
+
+    plt.figure()
+    plt.pcolormesh(data, cmap=cmap)
+    plt.colorbar(orientation='vertical')
+
+    cmap = plt.get_cmap('viridis', 128)[0.2:-0.3:16 * 1j]
+
+    plt.figure()
+    plt.pcolormesh(data, cmap=cmap)
+    plt.colorbar(orientation='vertical')
+
+    cmap = plt.get_cmap('viridis', 128)[25:90]
+
+    plt.figure()
+    plt.pcolormesh(data, cmap=cmap)
+    plt.colorbar(orientation='vertical')
+
+    cmap = plt.get_cmap('viridis', 128)[25:90:16 * 1j]
+
+    plt.figure()
+    plt.pcolormesh(data, cmap=cmap)
+    plt.colorbar(orientation='vertical')
+
+
```

Comment:
Not sure this requires image comparisons (which are repo-heavy).  Can you just test on the returned colormaps?

Reply:
Hi Jody! I hadn't thought of this, but it makes sense. Are the colormaps sitting somewhere that I can compare to, or... ?

Reply:
Hi Levi,  I'd just evaluate the colormap at 5 or 6 locations and assert that the values returned are what you say they should be.  That seems trivial but if someone mucks with your code they will have to also change the test and thats a good warning to double check what was done.